main/solution/backend/config/infra/cloudformation.yml

```diff
@@ -307,9 +307,158 @@ Resources:
+              - !GetAtt [IndexesDb, Arn]
+              - !GetAtt [CostApiCachesDb, Arn]
+              - !GetAtt [ProjectsDb, Arn]
+              - !GetAtt [AccountsDb, Arn]
+              - !GetAtt [KeyPairsDb, Arn]
+              - !Join ['', [!GetAtt [KeyPairsDb, Arn], '/index/*']]
+              - !GetAtt [StorageGatewayDb, Arn]
+              - !GetAtt [DsAccountsDb, Arn]
+              - !GetAtt [RoleAllocationsDb, Arn]
+              - !GetAtt [ResourceUsagesDb, Arn]
+          - Effect: Allow
+            Action:
+              - ssm:GetParameter
+            Resource:
+              - !Sub 'arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/${self:custom.settings.paramStoreRoot}/*'
+              - !Sub 'arn:aws:ssm:${AWS::Region}:*:parameter/aws/service/storagegateway/ami/FILE_S3/latest'
+          - Effect: Allow
+            Action:
+              - storagegateway:CreateNFSFileShare
+              - storagegateway:ListFileShares
+              - storagegateway:DescribeNFSFileShares
+              - storagegateway:UpdateNFSFileShare
+            Resource: !Sub 'arn:aws:storagegateway:${AWS::Region}:${AWS::AccountId}:gateway/*'
+          - Effect: Allow
+            Action:
+              - states:StartExecution
+            Resource:
+              - !Ref SMWorkflow
+          - Effect: Allow
+            Action:
+              - s3:GetObject*
+              - s3:AbortMultipartUpload
+              - s3:ListMultipartUploadParts
+              - s3:PutObject*
+              - s3:DeleteObject*
+            Resource:
+              - 'arn:aws:s3:::${self:custom.settings.studyDataBucketName}/*'
+              - 'arn:aws:s3:::${self:custom.settings.environmentsBootstrapBucketName}/*'
+              - 'arn:aws:s3:::${self:custom.settings.externalCfnTemplatesBucketName}/*'
+              - 'arn:aws:s3:::${self:custom.settings.envTypeConfigsBucketName}/*'
+          - Effect: Allow
+            Action:
+              - s3:ListBucket
+              - s3:GetBucketPolicy
+              - s3:PutBucketPolicy
+            Resource:
+              - 'arn:aws:s3:::${self:custom.settings.studyDataBucketName}'
+              - 'arn:aws:s3:::${self:custom.settings.environmentsBootstrapBucketName}'
+              - 'arn:aws:s3:::${self:custom.settings.envTypeConfigsBucketName}'
+          - Effect: Allow
+            Action:
+              - kms:GenerateDataKey
+              - kms:Decrypt
+              - kms:DescribeKey
+              - kms:Encrypt
+              - kms:ReEncrypt*
+              - kms:GetKeyPolicy
+              - kms:PutKeyPolicy
+            Resource:
+              # NOTE: Can't use '!GetAtt StudyDataEncryptionKey.Arn' due to
+              #   circular dependency in StudyDataEncryptionKey's KeyPolicy
+              - !Sub 'arn:aws:kms:${AWS::Region}:${AWS::AccountId}:alias/${self:custom.settings.studyDataKmsKeyAlias}'
+              - !Sub 'arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/*'
```

Comment:
Do we need access to all KMS keys? Which KMS keys do we need access to besides study KMS key?

Reply:
KMS key IDs are random strings, which is why they need to be a wildcard here.

Reply:
Synced up on Slack: there was a circular dependency which was causing us to add the wildcard. We found a workaround for that.